primitives/runtime/src/traits.rs

```diff
@@ -715,6 +703,47 @@ pub trait Hash:
 	fn trie_root(input: Vec<(Vec<u8>, Vec<u8>)>, state_version: StateVersion) -> Self::Output;
 }
 
+/// Super trait with all the attributes for a hashing output.
+pub trait HashOutput:
```

Comment:
While just a nitpick about the name that has been used. To me would be more correct to generally stick with one naming convention.

That is, seems there is some confusion in the naming convention adopted for hashers, hashes, and hash outputs.

* `sp-core` exports `Hasher` hash-db. We implement it for some types (e.g. `Keccak256`)
* We have this `Hash` trait here that technically is defining a hasher.
* We have `HashOutput` trait here that instrad is the `Hash`.

Can't we rename, just to be coherent these two traits?
`Hash` -> `Hesher`
`HashOutput` -> `Hash`

This would follow the convention adopted by the standard lib as well

Reply:
Personally I would also favor this, but this will for sure result in some very big diff :P

Reply:
I agree with you. Maybe you could rename them after this PR merged.

Reply:
BTW, there still is `Hashing` which should be `Hasher`?